internal/controller/apiextensions/claim/configurator.go

```diff
@@ -63,7 +65,15 @@ func ConfigureComposite(_ context.Context, cm resource.CompositeClaim, cp resour
 	// external name (even if that external name was empty) in order to ensure
 	// we don't try to rename anything after the fact.
 	if meta.WasCreated(cp) {
-		meta.SetExternalName(cp, en)
+		enCm := meta.GetExternalName(cm)
+		// if there is a mismatch between claim's external-name and composite's external-name
+		if enCm != "" && enCm != en {
+			return errors.New(errExternalNameMismatch)
+		}
+		// fix(2353): do not introduce a superfluous extern-name (empty external-names are treated as invalid)
```

Comment:
A few comment style nits - I believe the convention in this file is to:

1. Capitalize the first letter.
2. Wrap them at 80 chars.

Do you think we should establish a convention of linking to the PR number in comments using this `fix()` syntax? We've historically included full links to relevant issues/PRs when it's useful to do so.

Reply:
It may be useful to standardize a concise format for such purposes similar to a `TODO(<user>)`.